openmmtools/storage/storageinterface.py

```diff
@@ -0,0 +1,521 @@
+        """
+        Checks if the object can be used in the .write, .append, .fetch functions can be used. Once this is set, this
+        instance cannot be converted to a directory type.
+
+        Returns
+        -------
+        variable_pointer : None or storage_system specific unit of storage
+            Returns None if this instance is a directory, or if its functionality has not been determined yet (this
+                includes the variable not being assigned yet)
+            Returns the storage_system specific variable that this instance is bound to once assigned
+
+        """
+        return self._variable
+
+    @property
+    def directory(self):
+        """
+        Checks if the object can be used as a directory for accepting other SIDV objects. Once this is set, the
+        .write, .append, and .fetch functions are locked out.
+
+        Returns
+        -------
+        directory_pointer : None, or storage_system specific directory of storage
+            Returns None if this instance is a variable, or if its functionality has not been determined yet (this
+                includes the directory not being assigned yet)
+            Returns the storage_system specific directory that this instance is bound to once assigned
+
+        """
+        return self._directory
+
+    @property
+    def path(self):
+        """
+        Generate the complete path of this instance by getting its predecessor's path + itself. This is a cascading
+        function up the stack of SIDV's until the top level attached to the main SI instance is reached, then
+        reassembled on the way down.
+
+        Returns
+        -------
+        full_path : string
+            The complete path of this variable as it is seen on the storage file_name, returned as / separated values.
+
+        """
+        path = []
+        # Cascade the path
+        if self.predecessor is not None:
+            path = self.predecessor.path.split('/')  # Break path-like string to list to process
+        # Add self to the end
+        path.extend([self.name])  # Wrap in list or it iterates over the name chars
+        # Reduce to a path-like string
+        return '/'.join(path)
+
+    @property
+    def predecessor(self):
+        """
+        Give the parent SIDV to construct the full path on the way down
+
+        Returns
+        -------
+        predecessor : None or StorageInterfaceDirVar instance
+            Returns this instance's parent SIDV instance or None if it is the top level SIDV.
+        """
+        return self._predecessor
+
+    @property
+    def storage_driver(self):
+        """
+        Pointer to the object which actually handles read/write operations
+
+        Returns
+        -------
+        storage_driver :
+            Instance of the module which handles IO actions to specific storage type requested by storage_system
+            string at initialization.
+
+        """
+        return self._storage_driver
```

Comment:
I had originally created that variable as a protected name so you could not assign it as one of the variables on disk. You are right, I can just derive this and all it did was expose a hidden objects' property. I did a small refactor to remove this and just derive the object.